### PR TITLE
Julia transpose

### PR DESCRIPTION
**Pull request by matthijscox**
Removed the transpose() and some other minor updates

See: https://github.com/brainflow-dev/brainflow/issues/227

### CODE REVIEW

File: julia-package/brainflow/src/board_shim.jl [hunk 1]
```diff
@@ -49,12 +49,35 @@ end
...
+            catch
+                throw(BrainFlowError("you need to provide master board id to other_info field of BrainFlowInputParams", Integer(INVALID_ARGUMENTS_ERROR)))
+            end
+        end
+        new(master_id, id, JSON.json(params))
+    end
+
+end
+BoardShim(id::BoardIds, params::BrainFlowInputParams) = BoardShim(Integer(id), params)
+BoardShim(id) = BoardShim(id, BrainFlowInputParams())
```
Comment (Andrey1994) [inline julia-package/brainflow/src/board_shim.jl:72]: Lets remove this overload, its very important to keep API consistent across languages.

Also for almost all boards params needed, default params work only for OYmotion and Synthetic Board, it makes sense to "teach" users that they need to provide it

File: julia-package/brainflow/src/board_shim.jl [hunk 2]
```diff
@@ -49,12 +49,35 @@ end
...
+
 @brainflow_rethrow function get_sampling_rate(board_id::BoardIdType)
     val = Vector{Cint}(undef, 1)
     ccall((:get_sampling_rate, BOARD_CONTROLLER_INTERFACE), Cint, (Cint, Ptr{Cint}), Int32(board_id), val)
     value = val[1]
     return value
 end
+get_sampling_rate(board_shim::BoardShim) = get_sampling_rate(board_shim.master_board_id)
```
Comment (Andrey1994) [inline julia-package/brainflow/src/board_shim.jl:80]: for the same reason as above lets remove these overloaded methods, I understand the motivation for it, but the rule is to keep the same API between languages

File: julia-package/brainflow/src/data_filter.jl [hunk 3]
```diff
@@ -67,15 +73,29 @@ end
...
             data, length(data), Int32(sampling_rate), Float64(center_freq), Float64(band_width), Int32(order), Int32(filter_type), Float64(ripple))
     return
 end
 
-@brainflow_rethrow function perform_bandstop(data, sampling_rate::Integer, center_freq::Float64,
-    band_width::Float64, order::Integer, filter_type::Integer, ripple::Float64)
+@brainflow_rethrow function perform_bandstop(
+    data, 
+    sampling_rate::Integer,
+    center_freq::Float64 = 50.0, #Hz
```
Comment (Andrey1994) [inline julia-package/brainflow/src/data_filter.jl:93]: lets remove defaults here, for USA there must be different values and lets keep it up to the user to provide it

**Review by Andrey1994 — COMMENTED** [review]
lets also add smth about column major order to the UserAPI page in docs,

File: julia-package/brainflow/src/data_filter.jl (hunk 3 above)
Comment (matthijscox) [reply julia-package/brainflow/src/data_filter.jl:93]: Hmm... I had to spend time googling and asking you to figure out good defaults.
Is there a way to add them to the docs somehow?

File: julia-package/brainflow/src/board_shim.jl (hunk 1 above)
Comment (matthijscox) [reply julia-package/brainflow/src/board_shim.jl:72]: Alright

File: julia-package/brainflow/src/board_shim.jl (hunk 2 above)
Comment (matthijscox) [reply julia-package/brainflow/src/board_shim.jl:80]: Ok, though I hate writing more code in my scripts

File: julia-package/brainflow/src/data_filter.jl (hunk 3 above)
Comment (Andrey1994) [reply julia-package/brainflow/src/data_filter.jl:93]: maybe we can add methods like `perform_notch_eu` or `perform_notch_usa` to all languages?

Comment (matthijscox) [reply julia-package/brainflow/src/data_filter.jl:93]: I guess that's valid solution as well.

**Comment by matthijscox** [comment]
> lets also add smth about column major order to the UserAPI page in docs,

Done

I've also removed the MATLAB transpose, let's see what the CI says.

**Comment by Andrey1994** [comment]
"I've also removed the MATLAB transpose, let's see what the CI says."
for matlab there are no CI tests...
Its a paid software and not installed in CI containers\VMs, also I dont know how to use matlab from cmd, so even with a license its not simple to automate it.

Will need to test manually

**Comment by matthijscox** [comment]
Hmm, bloody MATLAB.

The installation instructions state I need to compile myself. Can't I just download them from the release page: https://github.com/brainflow-dev/brainflow/releases
However, the Julia compiled libs only have content of /lib folder, right? MATLAB instructions say there should be a /inc folder?

(I cant install Visual Studio at work, but I do have a MATLAB license there...)

**Comment by Andrey1994** [comment]
"However, the Julia compiled libs only have content of /lib folder, right? MATLAB instructions say there should be a /inc folder?

(I cant install Visual Studio at work, but I do have a MATLAB license there...)
"

yes, matlab also parses headers and inc folder is needed, I can send you an archieve in slack

**Comment by matthijscox** [comment]
Installed the MinGW compiler in MATLAB and used the /inc and /lib you sent me, I then got the following error upon calling a BrainFlow function (`BoardShim.set_log_file('brainflow.log');` in this case)

```
Error loading library intermediate output follows.
The actual error is at the end of this output.
*********
Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.30), passed through in regex; marked by <-- HERE in m/struct([^;,{(]*){ <-- HERE (.*?)\}([^;])*;/ at C:\Program Files\MATLAB\R2018b\toolbox\matlab\general\private\prototypes.pl line 912.

Failed to parse type '( __cdecl__ )) set_log_level ( int log_level' original input '( __cdecl__ )) set_log_level ( int log_level '
Found on line 79 of input from line 37 of file C:\\Localdata\\Repositories\\brainflow\\matlab-package\\brainflow\\inc\\board_controller.h
Error parsing argument for function __attribute__ function may be invalid.

Failed to parse type '( __cdecl__ )) set_log_file ( char * log_file' original input '( __cdecl__ )) set_log_file ( char * log_file '
Found on line 80 of input from line 38 of file C:\\Localdata\\Repositories\\brainflow\\matlab-package\\brainflow\\inc\\board_controller.h
Error parsing argument for function __attribute__ function may be invalid.
*********
Error using loadlibrary
Building BoardController_thunk_pcwin64 failed.  Compiler output is:
C:\ProgramData\MATLAB\SupportPackages\R2018b\3P.instrset\mingw_w64.instrset\bin\gcc -I"C:\Program Files\MATLAB\R2018b\extern\include" -fexceptions -fno-omit-frame-pointer -I"C:\Localdata\Repositories\brainflow\matlab-package\brainflow" -I"C:\Localdata\Repositories\brainflow\matlab-package\brainflow\inc" "BoardController_thunk_pcwin64.c" -o "BoardController_thunk_pcwin64.dll" -shared 
BoardController_thunk_pcwin64.c: In function 'int32':
BoardController_thunk_pcwin64.c:54:41: error: expected declaration specifiers before ')' token
 EXPORT_EXTERN_C int32_T int32(__cdecl__))set_log_level(intlog_levelThunk(void fcn(),const char *callstack,int stacksize)
                                         ^
BoardController_thunk_pcwin64.c:5:26: error: expected declaration specifiers before '__attribute__'
   #define DLL_EXPORT_SYM __declspec(dllexport)
                          ^
BoardController_thunk_pcwin64.c:19:25: note: in expansion of macro 'DLL_EXPORT_SYM'
 #define EXPORT_EXTERN_C DLL_EXPORT_SYM
                         ^~~~~~~~~~~~~~
BoardController_thunk_pcwin64.c:63:1: note: in expansion of macro 'EXPORT_EXTERN_C'
 EXPORT_EXTERN_C int32_T int32(__cdecl__))set_log_file(char*log_fileThunk(void fcn(),const char *callstack,int stacksize)
 ^~~~~~~~~~~~~~~
BoardController_thunk_pcwin64.c:54:25: warning: type of '__cdecl__' defaults to 'int' [-Wimplicit-int]
 EXPORT_EXTERN_C int32_T int32(__cdecl__))set_log_level(intlog_levelThunk(void fcn(),const char *callstack,int stacksize)
                         ^~~~~
BoardController_thunk_pcwin64.c:69:1: error: expected '{' at end of input
 }
 ^


Error in BoardShim.load_lib (line 14)
                    loadlibrary('BoardController.dll', 'board_controller.h', 'addheader', 'board_info_getter');

Error in BoardShim.set_log_file (line 49)
            lib_name = BoardShim.load_lib();

Error in BrainFlowGetData (line 4)
BoardShim.set_log_file('brainflow.log');
```

I'm starting to consider to revert the MATLAB transpose changes and just merge the Julia changes.

**Comment by Andrey1994** [comment]
I thought about it more, data returned from C to bindings(as well as passed from binding to C) is in row-major layout, so lets keep it as before

**Pull request closed without merging**